modules/nextflow/src/main/groovy/nextflow/splitter/JsonSplitter.groovy

```diff
@@ -217,9 +217,9 @@ class JsonSplitter extends AbstractTextSplitter {
                 break
             }
 
-            final map = Map.of(
-                 OBJECT_KEY, reader.nextName(),
-                 OBJECT_VALUE, fromJson(reader))
+            def map = new LinkedHashMap<>(1)
+            map[OBJECT_KEY] = reader.nextName()
+            map[OBJECT_VALUE] = fromJson(reader)
```

Comment:
This change fixed a weird bug where the JSON splitter would fail when parsing an object with a null value in it. It also makes the key order consistent which is good for the CI test 😄

Reply:
Ok, better to keep in its own commit for history purposes 👉 8ec14dd2